// graphql-2-json-schema/dist/test-utils.js
"use strict";
exports.__esModule = true;
var graphql_1 = require("graphql");
exports.getTodoSchemaIntrospection = function () {
    var schema = graphql_1.buildSchema("\n        \"A ToDo Object\"\n        type Todo {\n            \"A unique identifier\"\n            id: String!            \n            name: String!\n            completed: Boolean\n            color: Color\n        }\n\n        \"\"\"\n        A type that describes ToDoInputType. Its description might not\n        fit within the bounds of 80 width and so you want MULTILINE\n        \"\"\"\n        input TodoInputType {\n            name: String!\n            completed: Boolean\n            color: Color\n        }        \n        \n        enum Color {           \n          \"Red color\"          \n          RED\n          \"Green color\"\n          GREEN\n        }\n\n        type Query {\n            todo(\n                \"todo identifier\"\n                id: String!\n            ): Todo!\n            todos: [Todo!]!\n        }\n\n        type Mutation {\n            update_todo(id: String!, todo: TodoInputType!): Todo\n            create_todo(todo: TodoInputType!): Todo\n        }\n");
    var result = graphql_1.graphqlSync(schema, graphql_1.introspectionQuery);
    return {
        introspection: result.data,
        schema: schema
    };
};
exports.todoSchemaAsJsonSchema = {
    $schema: 'http://json-schema.org/draft-06/schema#',
    properties: {
        Query: {
            type: 'object',
            properties: {
                todo: {
                    type: 'object',
                    properties: {
                        arguments: {
                            type: 'object',
                            properties: {
                                id: { type: 'string', description: "todo identifier" }
                            },
                            required: ['id']
                        },
                        "return": {
                            $ref: '#/definitions/Todo'
                        }
                    },
                    required: []
                },
                todos: {
                    type: 'object',
                    properties: {
                        arguments: {
                            type: 'object',
                            properties: {},
                            required: []
                        },
                        "return": {
                            type: 'array',
                            items: { $ref: '#/definitions/Todo' }
                        }
                    },
                    required: []
                }
            },
            required: []
        },
        Mutation: {
            type: 'object',
            properties: {
                update_todo: {
                    type: 'object',
                    properties: {
                        arguments: {
                            type: 'object',
                            properties: {
                                id: { type: 'string' },
                                todo: { $ref: '#/definitions/TodoInputType' }
                            },
                            required: ['id', 'todo']
                        },
                        "return": {
                            $ref: '#/definitions/Todo'
                        }
                    },
                    required: []
                },
                create_todo: {
                    type: 'object',
                    properties: {
                        arguments: {
                            type: 'object',
                            properties: {
                                todo: { $ref: '#/definitions/TodoInputType' }
                            },
                            required: ['todo']
                        },
                        "return": {
                            $ref: '#/definitions/Todo'
                        }
                    },
                    required: []
                }
            }
        }
    },
    definitions: {
        'Todo': {
            type: 'object',
            description: "A ToDo Object",
            properties: {
                id: { type: 'string', description: "A unique identifier" },
                name: { type: 'string' },
                completed: { type: 'boolean' },
                color: { $ref: '#/definitions/Color' }
            },
            required: ['id', 'name']
        },
        'Color': {
            type: 'string',
            anyOf: [
                {
                    "enum": ['RED'],
                    title: 'Red color',
                    description: 'Red color'
                },
                {
                    "enum": ['GREEN'],
                    title: 'Green color',
                    description: 'Green color'
                }
            ]
        },
        'TodoInputType': {
            type: 'object',
            description: 'A type that describes ToDoInputType. Its description might not\nfit within the bounds of 80 width and so you want MULTILINE',
            properties: {
                name: { type: 'string' },
                completed: { type: 'boolean' },
                color: { $ref: '#/definitions/Color' }
            },
            required: ['name']
        }
    }
};
